public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -103,7 +103,7 @@ export class ElasticDatasource
     this.withCredentials = instanceSettings.withCredentials;
     this.url = instanceSettings.url!;
     this.name = instanceSettings.name;
-    this.index = instanceSettings.database ?? '';
+    this.index = instanceSettings.jsonData.index ?? instanceSettings.database ?? '';
```

Comment:
Can `instanceSettings.jsonData` be ever undefined? I am asking because 2 lines bellow we have `const settingsData = instanceSettings.jsonData || ({} as ElasticsearchOptions);` So I am thinking if we should moved this under line with  `const settingsData...`

Reply:
@ivanahuckova updated, good catch